[CI] DriverTests testThreadContext failing

This didn't reproduce for me but it's a newly added test and has already failed so is probably worth a look.

**Build scan:**
https://gradle-enterprise.elastic.co/s/jnsn6r7agwy3s/tests/:x-pack:plugin:esql:compute:test/org.elasticsearch.compute.operator.DriverTests/testThreadContext
**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.DriverTests.testThreadContext" -Dtests.seed=F277C2E26160F066 -Dtests.locale=ar-SA -Dtests.timezone=Australia/Currie -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.compute.operator.DriverTests&tests.test=testThreadContext
**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: "user1"
     but: was null

  at __randomizedtesting.SeedInfo.seed([F277C2E26160F066:6474F7E2FC8F4A3A]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.compute.operator.DriverTests.assertRunningWithRegularUser(DriverTests.java:89)
  at org.elasticsearch.compute.operator.DriverTests.lambda$testThreadContext$2(DriverTests.java:67)
  at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:128)
  at org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:310)
  at org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:230)
  at org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:133)
  at org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:108)
  at org.elasticsearch.compute.operator.DriverTests.testThreadContext(DriverTests.java:66)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/DriverTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/DriverTests.java
@@ -9,8 +9,6 @@ package org.elasticsearch.compute.operator;
 
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.ActionRunnable;
-import org.elasticsearch.action.support.PlainActionFuture;
-import org.elasticsearch.action.support.SubscribableListener;
 import org.elasticsearch.common.breaker.CircuitBreaker;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeValue;
@@ -34,12 +32,14 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
 import static org.hamcrest.Matchers.equalTo;
 
 public class DriverTests extends ESTestCase {
 
-    public void testThreadContext() {
+    public void testThreadContext() throws Exception {
         DriverContext driverContext = driverContext();
         ThreadPool threadPool = threadPool();
         try {
@@ -58,27 +58,28 @@ public class DriverTests extends ESTestCase {
                 outPages.add(page);
             }), () -> {});
             ThreadContext threadContext = threadPool.getThreadContext();
-            SubscribableListener<Void> future = new SubscribableListener<>();
+            CountDownLatch latch = new CountDownLatch(1);
             try (ThreadContext.StoredContext ignored = threadContext.stashContext()) {
                 threadContext.putHeader("user", "user1");
-                Driver.start(threadContext, threadPool.executor("esql"), driver, between(1, 1000), future);
+                Driver.start(threadContext, threadPool.executor("esql"), driver, between(1, 1000), ActionListener.running(() -> {
+                    try {
+                        assertRunningWithRegularUser(threadPool);
+                        assertThat(outPages, equalTo(inPages));
+                        Map<String, Set<String>> actualResponseHeaders = new HashMap<>();
+                        for (Map.Entry<String, List<String>> e : threadPool.getThreadContext().getResponseHeaders().entrySet()) {
+                            actualResponseHeaders.put(e.getKey(), Sets.newHashSet(e.getValue()));
+                        }
+                        Map<String, Set<String>> expectedResponseHeaders = new HashMap<>(warning1.warnings);
+                        for (Map.Entry<String, Set<String>> e : warning2.warnings.entrySet()) {
+                            expectedResponseHeaders.merge(e.getKey(), e.getValue(), Sets::union);
+                        }
+                        assertThat(actualResponseHeaders, equalTo(expectedResponseHeaders));
+                    } finally {
+                        latch.countDown();
+                    }
+                }));
             }
-            future.addListener(ActionListener.running(() -> {
-                assertRunningWithRegularUser(threadPool);
-                assertThat(outPages, equalTo(inPages));
-                Map<String, Set<String>> actualResponseHeaders = new HashMap<>();
-                for (Map.Entry<String, List<String>> e : threadPool.getThreadContext().getResponseHeaders().entrySet()) {
-                    actualResponseHeaders.put(e.getKey(), Sets.newHashSet(e.getValue()));
-                }
-                Map<String, Set<String>> expectedResponseHeaders = new HashMap<>(warning1.warnings);
-                for (Map.Entry<String, Set<String>> e : warning2.warnings.entrySet()) {
-                    expectedResponseHeaders.merge(e.getKey(), e.getValue(), Sets::union);
-                }
-                assertThat(actualResponseHeaders, equalTo(expectedResponseHeaders));
-            }));
-            PlainActionFuture<Void> completion = new PlainActionFuture<>();
-            future.addListener(completion);
-            completion.actionGet(TimeValue.timeValueSeconds(30));
+            assertTrue(latch.await(30, TimeUnit.SECONDS));
         } finally {
             terminate(threadPool);
         }

